x-pack/plugins/cases/public/components/use_push_to_service/index.tsx

```diff
@@ -69,71 +76,50 @@ export const usePushToService = ({
+    const errors: ErrorMessage[] = [];
 
     // these message require that the user do some sort of write action as a result of the message, readonly users won't
     // be able to perform such an action so let's not display the error to the user in that situation
     if (!userCanCrud) {
       return errors;
     }
 
-    if (actionLicense != null && !actionLicense.enabledInLicense) {
-      errors = [...errors, getLicenseError()];
+    /**
+     * We show only one message to the user depending the scenario. The messages have a priority of importance.
+     * Messages with higher priority are being shown first. The priority is defined by the order each message is returned.
+     *
+     * By priority of importance:
+     * 1. Show license error.
+     * 2. Show configuration error.
+     * 3. Show connector configuration error if the connector is set to none or no connectors have been created.
+     * 4. Show an error message if the connector has been deleted or the user does not have access to it.
+     * 5. Show case closed message.
+     */
+
+    if (hasLicenseError) {
+      return [getLicenseError()];
     }
 
-    if (connectors.length === 0 && connector.id === 'none' && !loadingLicense) {
-      errors = [
-        ...errors,
-        {
-          id: 'connector-missing-error',
-          title: '',
-          description: i18n.CONFIGURE_CONNECTOR,
-        },
-      ];
-    } else if (connector.id === 'none' && !loadingLicense) {
-      errors = [
-        ...errors,
-        {
-          id: 'connector-not-selected-error',
-          title: '',
-          description: i18n.CONFIGURE_CONNECTOR,
-        },
-      ];
-    } else if (!isValidConnector && !loadingLicense) {
-      errors = [
-        ...errors,
-        {
-          id: 'connector-deleted-error',
-          title: '',
-          description: (
-            <FormattedMessage
-              defaultMessage="The connector used to send updates to external service has been deleted. To update cases in external systems, select a different connector or create a new one."
-              id="xpack.cases.caseView.pushToServiceDisableByInvalidConnector"
-            />
-          ),
-          errorType: 'danger',
-        },
-      ];
+    if (actionLicense != null && !actionLicense.enabledInConfig) {
+      return [getKibanaConfigError()];
     }
-    if (caseStatus === CaseStatuses.closed) {
-      errors = [
-        ...errors,
-        {
-          id: CLOSED_CASE_PUSH_ERROR_ID,
-          title: '',
-          description: (
-            <FormattedMessage
-              defaultMessage="Closed cases cannot be sent to external systems. Reopen the case if you want to open or update it in an external system."
-              id="xpack.cases.caseView.pushToServiceDisableBecauseCaseClosedDescription"
-            />
-          ),
-        },
-      ];
+
+    if (connectors.length === 0 && connector.id === 'none' && !loadingLicense && !hasLicenseError) {
```

Comment:
Hmm so this if block and the one below are redundant I think.

I think we could get away with just doing:

```
if (connector.id === 'none' && !loadingLicense && !hasLicenseError) { return [getConnectorMissingInfo()]; }
```

right?

Reply:
Yes!